bundles/org.openhab.core.automation.module.script.rulesupport/src/main/java/org/openhab/core/automation/module/script/rulesupport/internal/loader/ScriptFileWatcher.java

```diff
@@ -252,9 +253,17 @@ private void checkFiles() {
         SortedSet<URL> reimportUrls = new TreeSet<URL>(new Comparator<URL>() {
             @Override
             public int compare(URL o1, URL o2) {
-                String f1 = o1.getPath();
-                String f2 = o2.getPath();
-                return String.CASE_INSENSITIVE_ORDER.compare(f1, f2);
+                Path path1 = Paths.get(o1.getPath());
+                Path path2 = Paths.get(o2.getPath());
+                String name1 = path1.getFileName().toString();
+                String name2 = path2.getFileName().toString();
+                int nameCompare = name1.compareToIgnoreCase(name2);
+                if (nameCompare != 0) {
+                    return nameCompare;
+                } else {
+                    int pathCompare = path1.getParent().toString().compareToIgnoreCase(path2.getParent().toString());
```

Comment:
AFAIK `getParent()` can return null, so this could result into a NPE.
But 
1. this file does not use nullness checks at all
2. I assume we can be sure that there will be a parent

Reply:
Noted. I will add nullness checks at some point when cleaning up automation. For this specific case, the parent cannot be null, since the scripts have to reside in /automation/jsr223/ for them to be loaded.